app/controllers/application_controller.rb

```diff
@@ -5,6 +5,137 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions 
+    set :session_secret, 'super secret'
   end
 
+  get '/' do 
+  	erb :index
+	end
+
+	get '/signup' do
+		if !session[:id].nil?
+			redirect '/tweets'
+		end
+		erb :"users/create_user"
+	end
+
+	post '/signup' do
+		if params[:username].empty?||params[:password].empty?||params[:email].empty?
+			redirect '/signup'
+		else
+			@user = User.create(params)
+			session[:id]= @user.id
+			redirect '/tweets'
+		end
+	end
+
+	get '/login' do
+		if !session[:id].nil?
+			redirect '/tweets'
+		end
+		erb :'/users/login'
+	end
+
+	post '/login' do
+		@user =User.find_by(:username=> params[:username])
+		
+		if @user && @user.authenticate(params[:password])
+			session[:id] = @user.id
+			redirect '/tweets'
+		else
+			redirect '/login'
+		end
+	end
+
+	get '/tweets' do
+		if session[:id].nil?
+			redirect '/login'
+		else
+
+			@tweets= Tweet.all
+
+			erb :"/tweets/tweets"
+			
+		end
+	end
+
+	get '/logout' do
+		#binding.pry
+		if !session[:id].nil?
+			session.clear
+			redirect '/login'
+		else
+			redirect '/'
+		end
+	end
+
+	get '/users/:user_slug' do
+		@user =User.find_by_slug(params[:user_slug])
+		erb :'users/show'
+	end
+
+	get '/tweets/new' do
+
+		if Helpers.is_logged_in?(session)
+			erb :'/tweets/create_tweet'
+		else
+			redirect '/login'
+		end
+	end
+
+	post '/tweets' do
+		redirect '/tweets/new' if params[:content].empty?
+		@user = Helpers.current_user(session)
+		@tweet = Tweet.create(content: params[:content], user_id: @user.id)
+		redirect "/tweets/#{@tweet.id}"
+	end
+
+	get '/tweets/:id' do
+		if session[:id].nil? #||session[:id]!=Tweet.find(params[:id]).user_id
+			redirect '/login'
+		end
+
+		@tweet= Tweet.find(params[:id])
+		erb :"/tweets/show_tweet"
+	end
+
+	get '/tweets/:id/edit' do 
+		if !Helpers.is_logged_in?(session)||Tweet.find(params[:id]).user_id!=session[:id]
+			redirect '/login'
+		end
+		@tweet= Tweet.find(params[:id])
+		erb :"/tweets/edit_tweet"
+	end
+
+	post '/tweets/:id' do
+		if params[:content]==""
+			redirect "/tweets/#{params[:id]}/edit"
+		end
+		@tweet = Tweet.find(params[:id])
+		@tweet.update(:content=> params[:content])
+		@tweet.save
+		redirect "/tweets/#{params[:id]}"
+	end
+
+	post '/tweets/:id/delete' do
+		#binding.pry
+		if session[:id]!=Tweet.find(params[:id]).user_id
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(:id => params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.